src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java

```diff
@@ -156,8 +156,10 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
         Result previousBuildResult = null;
         AbstractBuild<?, ?> previousBuild = build.getPreviousCompletedBuild();
 
-        if (forceNewIssue && currentBuildResult == Result.FAILURE) {
-            currentBuildResultFailure(build, listener, null, filename, vars);
+        if (forceNewIssue) {
+            if (currentBuildResult == Result.FAILURE) {
+                currentBuildResultFailure(build, listener, null, filename, vars);
+            }
         }
```

Comment:
@harshil-hbo What issue are you looking at here?  This change is actually identical to the previous code (you are already using `&&`, which is a conditional AND -- the right-hand-side of the operator is never evaluated if `forceNewIssue` is false.)  Also, even if the expression _was_ evaluated, it couldn't cause an exception -- comparing `NULL` to a value is always valid in Java.